src/PromiseCore.php

```diff
@@ -89,7 +89,7 @@ public function __construct(
             );
         }
 
-        if (get_resource_type($handle) !== 'curl') {
+        if (is_resource($handle) && get_resource_type($handle) !== 'curl') {
```

Comment:
check for php version.

maybe we should split all checks into something like this?
``` 
if (PHP_MAJOR_VERSION === 7) {
    both checks
} else {
    class check
}
```